commit 00c4bfd2adebd7d856198540395aa9dd97be5d54
Author: Steve Glendinning <steve.glendinning@shawell.net>
Date:   Thu Nov 8 06:26:21 2012 +0000

    usbnet: ratelimit kevent may have been dropped warnings
    
    when something goes wrong, a flood of these messages can be
    generated by usbnet (thousands per second).  This doesn't
    generally *help* the condition so this patch ratelimits the
    rate of their generation.
    
    There's an underlying problem in usbnet's kevent deferral
    mechanism which needs fixing, specifically that events *can*
    get dropped and not handled.  This patch doesn't address this,
    but just mitigates fallout caused by the current implemention.
    
    Signed-off-by: Steve Glendinning <steve.glendinning@shawell.net>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Index: linux-3.2.46/drivers/net/usb/usbnet.c
===================================================================
--- linux-3.2.46.orig/drivers/net/usb/usbnet.c	2013-05-30 13:35:16.000000000 +0000
+++ linux-3.2.46/drivers/net/usb/usbnet.c	2013-07-26 19:37:40.000000000 +0000
@@ -325,10 +325,12 @@
 void usbnet_defer_kevent (struct usbnet *dev, int work)
 {
 	set_bit (work, &dev->flags);
-	if (!schedule_work (&dev->kevent))
-		netdev_err(dev->net, "kevent %d may have been dropped\n", work);
-	else
+	if (!schedule_work (&dev->kevent)) {
+		if (net_ratelimit())
+			netdev_err(dev->net, "kevent %d may have been dropped\n", work);
+	} else {
 		netdev_dbg(dev->net, "kevent %d scheduled\n", work);
+	}
 }
 EXPORT_SYMBOL_GPL(usbnet_defer_kevent);
 
